### ISSUE TITLE
Question counter doesn't respect hidden variables

### ISSUE THREAD

**Issue by henryiii** [cookiecutter/cookiecutter#1929]
* Cookiecutter version: 2.3.0

### Description:

Cookiecutter now has colorful output and a little counter that tells a user how many questions are left.

However, the counter doesn't take into account hidden / computed values. In the cookiecutter below, it thinks there are 14 questions, when there are only 9 that get asked. It should remove underscore variables when counting the total.

(Also, this counter will complicate #1914 a bit, but probably manageable)

### What I've run:

```console
$ cookiecutter gh:scientific-python/cookie
You've downloaded /Users/henryschreiner/.cookiecutters/cookie before. Is it okay to delete and re-download it? [y/n] (y):
  [1/14] The name of your project (package):
  [2/14] The name of your (GitHub?) org (org):
  [3/14] The url to your GitHub or GitLab repository (https://github.com/org/package):
  [4/14] Your name (My Name):
  [5/14] Your email (me@email.com):
  [6/14] A short description of your project (A great package.):
  [7/14] Select a license
    1 - BSD
    2 - Apache
    3 - MIT
    Choose from [1/2/3] (1):
  [8/14] Choose a build backend
    1 - Hatchling                      - Pure Python (recommended)
    2 - Flit-core                      - Pure Python (minimal)
    3 - PDM-backend                    - Pure Python
    4 - Whey                           - Pure Python
    5 - Poetry                         - Pure Python
    6 - Setuptools with pyproject.toml - Pure Python
    7 - Setuptools with setup.py       - Pure Python
    8 - Setuptools and pybind11        - Compiled C++
    9 - Scikit-build-core              - Compiled C++ (recommended)
    10 - Meson-python                  - Compiled C++ (also good)
    11 - Maturin                       - Compiled Rust (recommended)
    Choose from [1/2/3/4/5/6/7/8/9/10/11] (1):
  [9/14] Use version control for versioning [y/n] (y): n
$
```

**Comment by ericsouza** [comment]
+1

**Comment by ericof** [comment]
It was fixed on release 2.3.1
